### ISSUE TITLE
As a smith I want to port the type system judgments to Java

### ISSUE THREAD

**Issue by mor-n4** [eclipse/n4js#1167]
Due to the findings in recent performance investigations, we decided to start porting the type system judgments to Java.

At first this is done on an experimental level. Only after further investigation and discussion, if deemed necessary/desirable, it will be finalized and merged to master.

ACs:
1. port type system judgments to Java,
2. remove all obsolete files and dependencies.

NOTE: probably this is a good opportunity for clean up beyond just porting the code; however, such clean up is not part of this user story.
